tests/integrations/pyramid/test_pyramid.py

```diff
@@ -380,6 +380,16 @@ def authenticated_userid(self, request):
 
     assert len(events) == 1
 
+    # In `authenticated_userid` there used to be a call to `logging.error`. This would print this error in the
+    # event processor of the Pyramid integration and the logging integration would capture this and send it to Sentry.
+    # This is not possible anymore, because capturing that error in the logging integration would again run all the
+    # event processors (from the global, isolation and current scope) and thus would again run the same pyramid
+    # event processor that raised the error in the first place, leading on an infinite loop.
+    # This test here is now deactivated and always passes, but it is kept here to document the problem.
+    # This change in behavior is also mentioned in the migration documentation for Python SDK 2.0
```

Comment:
Ah, so we'd get a Pyramid error, then the Pyramid event processor would run and since we're trying to add user info onto the event in it, it would trigger the log message, leading to that being captured and all the event processors being run again, etc. What a mess.

I think we should in general just ignore errors in event processors tbh. Like, it should be a best-effort "I'm adding all potentially useful info to this event but no biggie if it doesn't work out". Maybe we can just wrap that stuff in `capture_internal_exceptions`, because that's ultimately what it is: we caused an exception while trying to do something Sentry-related, doesn't matter that it happened in non-SDK code.

Reply:
Oh wait, it apparently is already wrapped in `capture_internal_exceptions` 🤔 Then I don't get why we're still capturing it.

Reply:
We fixed the test but if something like this happens in a user's `authenticated_userid`, will the SDK blow up on them?